### DISCUSSION TITLE
Getting some linking errors while using some functions of magickwand.

### DISCUSSION THREAD

**Discussion by pme9** [ImageMagick/ImageMagick#5030] [Help]
I am trying to read image by magickwand, like below:

MagickWand* magick_wand;
MagickBooleanType status = MagickReadImage(magick_wand, filepath); 

but I'm getting these linking errors and some warnings:
Errors:
1>ReadPhotoshop.obj : error LNK2019: unresolved external symbol __imp_DestroyPixelIterator referenced in function "public: int __cdecl CReadPhotoshop::GetPixelDataIM(void)" (?GetPixelDataIM@CReadPhotoshop@@QEAAHXZ)
1>ReadPhotoshop.obj : error LNK2019: unresolved external symbol __imp_NewPixelIterator referenced in function "public: int __cdecl CReadPhotoshop::GetPixelDataIM(void)" (?GetPixelDataIM@CReadPhotoshop@@QEAAHXZ)
1>ReadPhotoshop.obj : error LNK2019: unresolved external symbol __imp_PixelGetNextIteratorRow referenced in function "public: int __cdecl CReadPhotoshop::GetPixelDataIM(void)" (?GetPixelDataIM@CReadPhotoshop@@QEAAHXZ)

Warnings:
1>LINK : warning LNK4217: symbol 'MagickWandGenesis' defined in 'CORE_DB_MagickWand_.lib(magick-wand.obj)' is imported by 'ReadPhotoshop.obj' in function '"public: int __cdecl CReadPhotoshop::GetPixelDataIM(void)" (?GetPixelDataIM@CReadPhotoshop@@QEAAHXZ)'
1>LINK : warning LNK4217: symbol 'MagickWandTerminus' defined in 'CORE_DB_MagickWand_.lib(magick-wand.obj)' is imported by 'ReadPhotoshop.obj' in function '"public: int __cdecl CReadPhotoshop::GetPixelDataIM(void)" (?GetPixelDataIM@CReadPhotoshop@@QEAAHXZ)'

do I need to add any library to access this functions properly? 

Note: I already added CORE_DB_MagickWand_.lib

**Comment by snibgo** [comment]
What version of IM? What toolchain do you use to build IM and your program?

From "DB", I suspect you are using MSVC, and this is the "debug" build. Comments in this forum suggest that "debug" shouldn't be used.

**Reply by pme9** [reply]
I'm using version 7 of IM. 
I have used visual studio 19 on windows platform to build IM and created Static Multi-threaded DLL of IM.
And yes, I'm using debug build.

Below is a code, which is throwing these linking errors:

	      MagickWandGenesis();
              MagickWand* magick_wand = NewMagickWand();
              const char* filepath = (LPCSTR)m_pPhotoshopApp->GetFilePath().GetBuffer();
              MagickBooleanType status = MagickReadImage(magick_wand, filepath);
 
              if (status == MagickFalse)
                     return;
              PixelIterator* iterator = MagickCore::NewPixelIterator(magick_wand);
              if (iterator == (PixelIterator*)NULL)
                     return; 

I have also added:
CORE_DB_Magick++_.lib
CORE_DB_MagickCore_.lib
CORE_DB_MagickWand_.lib
